Lmr 55 allfeedbacks

Daria-pugacheva opened the pull request:
ВНИМАНИЕ: В эту ветку объединила две ранее созданные ветки по этой задаче, потому что уже работаем над ними совместно (раньше, так вышло ,что нужны были разные ветки. а сейчас удобнее все делать в одной). Т.е PR , который LMR-55-feedbackaboutrenter уже не смотрим (он здесь внутри)
В ревьюеры пригласила в том числе Дмитрия , т.к. от него жду дополнительное поле в InstrumentRentDto с айдишником арендатора (написала TODO-шку), и Алексея  т.к. от него жду дополнительное поле в InstrumentInfoDto с айдишником владельца(написала TODO-шку).
В целом, эту ветку уже Татьяна и Николай отсматривали (когда это были две разные ветки). Сейчас функционал отправки отзывов срабатывает, НО! У меня серьезный затык с тем, чтобы положить в поля ng-model имеющиеся на фронте айдишники инструмента, владельца, арендатора и автора отзыва.  Пока я туда вручную запихнула цифры. 
(пока не готовы изменения в DTO-шки, на фронте не хватает, конечно айдишников арендатора и владельца, но все равно надо понять, как их класть в модель...).
В предпоследнем коммите я написала, где споткнулась, но дублирую сюда, т.к. решим эту дилемму и все будет летать отлично. 
Может, кто подскажет? Я пробовала и так и эдак, не получается. Вот что пыталась предпринять:
1. $scope.tool_comment.instrumentId = $routeParams.toolId; - это я, когда наброски делала, нафантазировала, но такой вариант не рабочий.
2. В самой html добавить в Input-поле атрибут valuе чтобы он по умолчанию заполнялся нашими данными, не получается (я так понимаю, поскольку у нас там уже есть ng-model  и она перекрывает)
3. На странице js просто создала объекты типа $scope.tool_comment={userId: "1", instrumentId: "2"}; - но тут только ручной ввод именно значений и не нашла как в json-нину передать нужные мне переменные $rootScope.myUserIdFromProfile, $routeParams.toolId и т.д. вместо цифр
4. Была еще мысль уж отправлять с фронта запрос на бэк, в котором передавать два айдишника в PathVariable и ответом как бы эхо получать объект с названиями полей, как нужно нам в модели, заполненными этими значениями и тогда как бы мы с бэка получаем модель с двумя полями. Но это какая-то дичь... Все-таки хотелось бы прямо на фронте засовывать в ng-model известные нам поля.

Daria-pugacheva committed:
Накидала функционал отправки отзывов владельца об арендаторе с учетом того, что на бэке по этому функционалу предложил сделать Артем.
1. Сделана отдельная страница для написания отзывов.
Пока нет возможности запустить посмотреть, как функционирует, сомневаюсь в двух моментах:
 - корректно ли я заполняю по умолчанию два поля модели - это строки 4 и 5 в feedback-about-renter-page.js
 - корректно ли отправляю нужных хэдер "about"  - строки 11-13 в feedback-about-renter-page.js
2. Поправила в соответствии с логикой методы в файлах с tool-history
3. Оставила TODO запрос в InstrumentRentDto, чтобы в нем появилось еще поле с айдишником арендатора (он нам нужен для отправки отзывов по предложенной Артемом логике)
4. И в UserController тоже оставила предложение сделать метод, отдающий данные о юзере по айдишнику
5. В создала глобальную переменную $rootScope.myUserIdFromProfile для реализации  отправок отзывов теперь сохраняю айдишник текущего пользователя.

Daria-pugacheva committed:
Merge branch 'LMR-55-feddbackaboutrenter' into LMR-55-feedbackAboutInstrumentAndOwner

# Conflicts:
#	source/core/src/main/resources/static/profile/profile.html
#	source/core/src/main/resources/static/profile/profile.js

Daria-pugacheva committed:
Доработан функционал по отправке отзывов об инструменте и его владельце (по отдельноти  и вместе) в соответствии с логикой на бэке, которая добавлена Артемом в ветке LMR-12
1. В InstrumentInfoDto написала TODO с просьбой добавить поле с айдишником владельца. Очень надо для реализации функционала отправки отзывов.
2. Пока не запускала приложение для проверки функциональности, так что, как и в ветке LMR-55-feedbackaboutRenter, пока под сомнением корректность ввода полей ng-model с необходимыми айдишниками и отправки хэдера на бэк
3. Думаю, что хорошо бы еще прямо на фронте добавить валидацию на поле ввода оценки (grade). Но я это пока не реализовала.

Daria-pugacheva committed:
Небольша доработка : в profile.js добавила $rootScope.myUserIdFromProfile

Daria-pugacheva committed:
Update source/core/src/main/resources/static/feedback-about-renter-page/feedback-about-renter-page.js

Co-authored-by: Nikolay Balashov <46228837+siporqueno@users.noreply.github.com>

Daria-pugacheva committed:
Update source/core/src/main/resources/static/feedback-about-renter-page/feedback-about-renter-page.js

Co-authored-by: Nikolay Balashov <46228837+siporqueno@users.noreply.github.com>

Daria-pugacheva committed:
Merge branch 'develop' into LMR-55-feddbackaboutrenter

Daria-pugacheva committed:
Merge remote-tracking branch 'origin/develop' into LMR-55-feddbackaboutrenter

# Conflicts:
#	source/core/src/main/resources/static/profile/profile.html
#	source/core/src/main/resources/static/profile/profile.js

Daria-pugacheva committed:
Небольша доработка : в profile.js добавила $rootScope в используемые зависимости.
И смержила ветку с текущей develop для удобства

Daria-pugacheva committed:
Merge remote-tracking branch 'origin/LMR-55-feddbackaboutrenter' into LMR-55-feddbackaboutrenter

# Conflicts:
#	source/core/src/main/resources/static/profile/profile.html
#	source/core/src/main/resources/static/profile/profile.js

Daria-pugacheva committed:
Крошечные правки для тестирования работоспособности (уберу их, когда проверю)

Daria-pugacheva committed:
Merge remote-tracking branch 'origin/develop' into LMR-55-feedbackAboutInstrumentAndOwner

Daria-pugacheva committed:
Смержила ветку со свежей develop +  поправила названия header-ов в функциях отправки отзывов и заменила в этих функциях param на data

Daria-pugacheva committed:
В index.js добавлены обратно пропавшие две скобки в троке 54

Daria-pugacheva committed:
Merge branch 'LMR-55-feddbackaboutrenter' into LMR-55-feedbackAboutInstrumentAndOwner

# Conflicts:
#	source/core/src/main/resources/static/profile/profile.html
#	source/core/src/main/resources/static/profile/profile.js

Daria-pugacheva committed:
Залила в эту ветку ветвь с отзывами об арендаторе + сделала пока ручное заполнение двух полей в ng-model произвольными цифрами с id инструмента, аредатора, владельца и автора отзыва.
Есть желание продолжить работать над задачей 55 в этой одной ветке.
Пока у меня вот прямо лютый затык с вопросом, как передавать данные про  id инструмента, аредатора, владельца и автора отзыва в соответствующие поля ng-model. Может, кто подскажет? Я пробовала и так и эдак, не получается. Вот что пыталась предпринять:
1. $scope.tool_comment.instrumentId = $routeParams.toolId; - это я, когда наброски делала, нафантазировала, но такой вариант не рабочий.
2. В самой html добваить в Input-поле атрибут valuе чтобы он по умолчанию заполнялся нашими данными, не получается (я так понимаю, поскольку у нас там уже есть ng-model  и она перекрывает)
3. На странице js просто создала объекты типа $scope.tool_comment={userId: "1", instrumentId: "2"}; - но тут только ручной ввод именно значений и не нашла как в json-нину передать нужные мне переменные $rootScope.myUserIdFromProfile, $routeParams.toolId и т.д. вместо цифр
4. Была еще мысль уж отправлять с фронта запрос на бэк, в котором передавать два айдишника в PathVariable и ответом как бы эхо получать обхект с названиями полей, как нужно нам в модели, заполенными этими значениями и тогда как бы мы с бэка полчаем модель с двумя полями. Но это какая-то дичь... Все-таки хотелось бы прямо на фронте засовывать в ng-model известные нам поля.

Daria-pugacheva committed:
Переименовала ветку, чтобы было понятно, что в ней все варианты отправки комментариев

Daria-pugacheva committed:
Добавила поле для ввода оценки по 5-бальной шкале, где не было

Commit:
LMR-55-ALLFEEDBACKS правка InstrumentRentDto

siporqueno committed:
Used rootScope to share data between two controllers.

Daria-pugacheva committed:
Merge branch 'LMR-55-ALLFEEDBACKS' of https://github.com/siporqueno/let-me-rent into LMR-55-ALLFEEDBACKS

Daria-pugacheva committed:
Чуть подправила форму вывода заказов в профиле и на странице истории (согласно свежей редакции InstrumentRentDto)

siporqueno committed:
Corrected navToFeedbackPage and it works! Yesterday night by mistake amended navToToolHistory

Daria-pugacheva committed:
Merge remote-tracking branch 'origin/develop' into LMR-55-ALLFEEDBACKS

# Conflicts:
#	source/core/src/main/java/ru/letmerent/core/dto/InstrumentInfoDto.java

Daria-pugacheva committed:
Смержила с develop, где обновился InstrumentInfoDto

Daria-pugacheva committed:
Смержила с develop, где обновился InstrumentInfoDto

Daria-pugacheva committed:
Смержила с develop, где обновился InstrumentInfoDto

Daria-pugacheva committed:
Merge pull request #68 from siporqueno/LMR-55-feedbackAboutInstrumentAndOwner

Lmr 55 feedback about instrument and owner

Daria-pugacheva committed:
Merge pull request #65 from siporqueno/LMR-55-ALLFEEDBACKS-data-sharing

Вариант как передавать айдишники через rootScope

Daria-pugacheva committed:
Работает весь функционал :) (только, когда бэк с отдачей заказов появится, надо будет еще раскомментировать соответствующий код.
1. Убрала лишние $routeParams, поскольку передаем нужные айдишники теперь единообразно через $rootScope.
2. Оживила страницу отзывов об арендаторе в соответствии с предложенной методикой передачи полей в ng-model
3. Почистила код от лишних закомментированных строк.

По-моему, можно с чистым сердцем мержить в девелоп :)

tatiananosova commented:
Всё работает, единственное что, если собственник брал в аренду свой же инструмент, то он может оставить о себе же отзыв.

Daria-pugacheva commented:
По поводу отзыва собственника о своем же инструменте, кстати, отличное замечание. Сделаю соответствующую проверку на фронте, пожалуй, сразу.

Daria-pugacheva committed:
1. Теперь кнопка отзывов об инструменте не активна в случае, если отзыв пытается оставить владелец этого инструмента (в т.ч. вылетает alert, чтобы пользователь понимал, почему не получается воспользоваться кнопкой)
2. На форме отзывов заменила поле ввода значений рейтинга на селект, чтобы выбирались значения в диапазоне от 1 до 5 + поставила проверку, что поле описания и рейтинга должны быть заполнены, чтобы отправить отзыв (а то есть риск, что кто-то легким движением руки наотправляет в базу пустых отзывов и это будет так себе выглядеть на странице инструмента, куда выводятся отзывы).  Единственное в этой части есть небольшой косяк: если ничего не вводить и нажать кнопку отправить, то выводится alert, что надо заполнить поля; если ввести только описание, а рейтинг не поставить (или наоборот), то тоже вылетает alert. А если сначала ввести описание без рейтинга, а потом ввести рейтинг, а описание затереть, то отзыв улетает.... В теории, вряд ли пользователи будут так развлекаться, но такое есть (и пока не докрутила, как это разрулить: только увидела этот момент. Но все равно так лучше, чем без валидации вообще)

Daria-pugacheva committed:
1. Теперь кнопка отзывов об инструменте не активна в случае, если отзыв пытается оставить владелец этого инструмента (в т.ч. вылетает alert, чтобы пользователь понимал, почему не получается воспользоваться кнопкой)
2. На форме отзывов заменила поле ввода значений рейтинга на селект, чтобы выбирались значения в диапазоне от 1 до 5 + поставила проверку, что поле описания и рейтинга должны быть заполнены, чтобы отправить отзыв (а то есть риск, что кто-то легким движением руки наотправляет в базу пустых отзывов и это будет так себе выглядеть на странице инструмента, куда выводятся отзывы).  Единственное в этой части есть небольшой косяк: если ничего не вводить и нажать кнопку отправить, то выводится alert, что надо заполнить поля; если ввести только описание, а рейтинг не поставить (или наоборот), то тоже вылетает alert. А если сначала ввести описание без рейтинга, а потом ввести рейтинг, а описание затереть, то отзыв улетает.... В теории, вряд ли пользователи будут так развлекаться, но такое есть (и пока не докрутила, как это разрулить: только увидела этот момент. Но все равно так лучше, чем без валидации вообще)

Daria-pugacheva committed:
Merge pull request #72 from siporqueno/LMR-55-feedbackAboutInstrumentAndOwner

Слияние из-за переименования ветки

siporqueno commented on source/core/src/main/resources/static/feedback-page/feedback-page.html:
```diff
@@ -20,22 +20,53 @@ <h3>Владелец инструмента:</h3>
+                <option value="1">1</option>
+                <option value="2">2</option>
+                <option value="3">3</option>
+                <option value="3">4</option>
+                <option value="3">5</option>
```
```suggestion
                <option value="4">4</option>
                <option value="5">5</option>
```

siporqueno commented on source/core/src/main/resources/static/feedback-page/feedback-page.html:
```diff
@@ -20,22 +20,53 @@ <h3>Владелец инструмента:</h3>
+                <option value="1">1</option>
+                <option value="2">2</option>
+                <option value="3">3</option>
+                <option value="3">4</option>
+                <option value="3">5</option>
```
```suggestion
                <option value="4">4</option>
                <option value="5">5</option>
```

siporqueno reviewed (approved):
Мне кажется, что страницу  для отзывов об инструменте и арендаторе feedback-page.html можно упростить: обе формы объединить в одну, три кнопки в одну. Сверху селект на два поля в котором выбираешь отзыв об инструменте или об арендаторе. Возврат на эту же страницу после отправки как вариант.

Daria-pugacheva committed:
Update source/core/src/main/resources/static/feedback-page/feedback-page.html

да, проглядела пару значений. Спасибо!

Co-authored-by: Nikolay Balashov <46228837+siporqueno@users.noreply.github.com>

Daria-pugacheva committed:
Оптимизирована логика оставления отзыва об инструменте: убрала отдельную форму для отзыва о владельце (при желании мнение о владельце пользователь пишеть в поле для отзыва об инструменте)

Daria-pugacheva committed:
Merge pull request #74 from siporqueno/LMR-55-feedbackAboutInstrumentAndOwner

Оптимизирована логика оставления отзыва об инструменте

Daria-pugacheva committed:
Merge branch 'develop' into LMR-55-ALLFEEDBACKS
